hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,915 @@
+    def get_inventory(self):
+        """
+        Retrieves the inventory details for all products with a SKU.
+
+        This function searches for all products in the Odoo database that have a SKU
+        (default_code not null) and returns their inventory details in a JSON response.
+
+        JSON response:
+            - message (str): A message indicating the action performed.
+            - inventory_data (list of dict): A list of dictionaries containing the inventory details for each product, including:
+                - name (str): The name of the product.
+                - default_code (str): The SKU of the product.
+                - qty_available (float): The quantity of the product available.
+                - virtual_available (float): The virtual quantity of the product available.
+
+        Returns:
+            dict: A dictionary with a message and the inventory details of the products.
+        """
+        products = request.env["product.product"].search(
+            [("default_code", "!=", False)]
+        )
+
+        product_fields = ["name", "default_code", "qty_available", "virtual_available"]
+        inventory_list = products.read(product_fields)
+
+        return {
+            "message": "Inventory data retrieved",
+            "inventory_data": inventory_list,
+        }
+
+    @route("/get_inventory_by_sku", methods=["POST"], type="json", auth="user")
+    def get_inventory_by_sku(self):
+        """
+        Retrieves the inventory details for a product based on its SKU.
+
+        This function searches for a product in the Odoo database using the provided SKU
+        and returns the inventory details in a JSON response.
+
+        JSON request body:
+            - sku (str): The SKU of the product to search for.
+
+        JSON response:
+            - message (str): A message indicating the action performed.
+            - inventory_data (list of dict): A list of dictionaries containing the product's inventory details, including:
+                - name (str): The name of the product.
+                - default_code (str): The SKU of the product.
+                - qty_available (float): The quantity of the product available.
+                - virtual_available (float): The virtual quantity of the product available.
+
+        Returns:
+            dict: A dictionary with a message and the inventory details of the product.
+        """
+        sku = request.jsonrequest.get("sku")
+        product = request.env["product.product"].search([("default_code", "=", sku)])
+        product_fields = ["name", "default_code", "qty_available", "virtual_available"]
+        data_product = product.read(product_fields)
+
+        return {
+            "message": "Inventory data retrieved",
+            "inventory_data": data_product,
+        }
+
+    @route("/get_states_mexico", methods=["GET"], type="json", auth="user")
+    def get_states_mexico(self):
```

Comment:
I think you can retrive all, and filter them at django

similar to 
https://github.com/odoo/odoo/blob/ab01f91913f32e11cc20287f719933849f0949eb/addons/website/controllers/main.py#L215-L218